pygit2/repository.py

```diff
@@ -996,6 +998,54 @@ def set_ident(self, name, email):
         check_error(err)
 
 
+class Branches(object):
+    def __init__(self, repository, flag=GIT_BRANCH_LOCAL | GIT_BRANCH_REMOTE):
+        self._repository = repository
+        self._flag = flag
+
+        if flag == GIT_BRANCH_LOCAL | GIT_BRANCH_REMOTE:
+            self.local = Branches(repository, flag=GIT_BRANCH_LOCAL)
+            self.remote = Branches(repository, flag=GIT_BRANCH_REMOTE)
+
+    def __getitem__(self, name):
+        branch = None
+        if self._flag & GIT_BRANCH_LOCAL:
+            branch = self._repository.lookup_branch(name, GIT_BRANCH_LOCAL)
+
+        if branch is None and self._flag & GIT_BRANCH_REMOTE:
+            branch = self._repository.lookup_branch(name, GIT_BRANCH_REMOTE)
```

Comment:
This does not work. `lookup_branch` can take `GIT_BRANCH_REMOTE` or nothing.
See:
https://github.com/libgit2/libgit2/blob/master/src/branch.c#L353
https://github.com/libgit2/libgit2/blob/master/src/branch.c#L31

So as far as I can tell, this is needed.